packages/data/addon/utils/adapter.ts

```diff
@@ -20,11 +21,25 @@ export function getDataSource<T extends keyof DataSourceRegistry>(dataSourceName
   const {
     navi: { dataSources },
   } = config;
-  const dataSource = dataSources.find(({ name }) => name === dataSourceName);
-  if (dataSource) {
-    return dataSource as DataSourceRegistry[T];
+  const [lookupName, lookupNamespace] = dataSourceName.split('.');
+  const dataSource = dataSources.find(({ name }) => name === lookupName);
+  if (!dataSource) {
+    throw new Error(`Datasource "${lookupName}" should be configured in the navi environment`);
   }
-  throw new Error(`Datasource ${dataSourceName} should be configured in the navi environment`);
+  if (lookupNamespace) {
+    const namespace = dataSource.namespaces?.find(({ name }) => name === lookupNamespace);
+    if (!namespace) {
```

Comment:
Do we store `elideOne.default` as a data source?

Reply:
no, just `elideOne`